src/main/java/io/supertokens/webserver/api/bulkimport/ImportUserAPI.java

```diff
@@ -0,0 +1,103 @@
+import io.supertokens.pluginInterface.Storage;
+import io.supertokens.pluginInterface.StorageUtils;
+import io.supertokens.pluginInterface.authRecipe.AuthRecipeUserInfo;
+import io.supertokens.pluginInterface.bulkimport.BulkImportUser;
+import io.supertokens.pluginInterface.exceptions.DbInitException;
+import io.supertokens.pluginInterface.exceptions.InvalidConfigException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.multitenancy.AppIdentifier;
+import io.supertokens.pluginInterface.multitenancy.exceptions.TenantOrAppNotFoundException;
+import io.supertokens.storageLayer.StorageLayer;
+import io.supertokens.utils.Utils;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class ImportUserAPI extends WebserverAPI {
+  public ImportUserAPI(Main main) {
+    super(main, "");
+  }
+
+  @Override
+  public String getPath() {
+    return "/bulk-import/import";
+  }
+
+  @Override
+  protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    // API is app specific
+
+    if (StorageLayer.isInMemDb(main)) {
+      throw new ServletException(new BadRequestException("This API is not supported in the in-memory database."));
+    }
+
+    JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+    JsonObject jsonUser = InputParser.parseJsonObjectOrThrowError(input, "user", false);
+
+    AppIdentifier appIdentifier = null;
+    Storage storage = null;
+    String[] allUserRoles = null;
+
+    try {
+      appIdentifier = getAppIdentifier(req);
+      storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+      allUserRoles = StorageUtils.getUserRolesStorage(storage).getRoles(appIdentifier);
+    } catch (TenantOrAppNotFoundException | BadPermissionException | StorageQueryException e) {
+      throw new ServletException(e);
+    }
+
+    BulkImportUserUtils bulkImportUserUtils = new BulkImportUserUtils(allUserRoles);
+
+    try {
+      BulkImportUser user = bulkImportUserUtils.createBulkImportUserFromJSON(main, appIdentifier, jsonUser,
+          Utils.getUUID());
+
+      AuthRecipeUserInfo importedUser = BulkImport.importUser(main, appIdentifier, user);
+
+      JsonObject result = new JsonObject();
+      result.addProperty("status", "OK");
+      result.add("user",  importedUser.toJson());
+      super.sendJsonResponse(200, result, resp);
+    } catch (io.supertokens.bulkimport.exceptions.InvalidBulkImportDataException e) {
+      JsonArray errors = e.errors.stream()
+          .map(JsonPrimitive::new)
+          .collect(JsonArray::new, JsonArray::add, JsonArray::addAll);
+
+      JsonObject errorResponseJson = new JsonObject();
+      errorResponseJson.add("errors", errors);
```

Comment:
what is an example of how this output looks like?

Reply:
```json
{
  "errors": [
    "Role role1 does not exist.",
    "Role role2 does not exist.",
    "Invalid tenantId: t1 for emailpassword recipe.",
    "Invalid tenantId: t1 for thirdparty recipe.",
    "Invalid tenantId: t1 for passwordless recipe."
  ]
}
```